include/circt/Dialect/Pipeline/PipelineOps.td

```diff
@@ -207,16 +207,31 @@ def StageOp : Op<Pipeline_Dialect, "stage", [
     to communicate:
     1. which stage (block) to transition to next
     2. which registers to build at this stage boundary
-    3. which values to pass through to the next stage without registering
+    3. An optional hierarchy of boolean values to be used for clock gates for
```

Comment:
the order of the new 3 and 4 is reversed wrt the builders. I would suggest reversing the argument order in the builder.